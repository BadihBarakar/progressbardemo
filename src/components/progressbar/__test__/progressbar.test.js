import React from "react";
import ReactDOM from "react-dom";
import ProgressBar from "../progressbar";

import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

it("render progressbar successfully", () => {
  // test that the component will render successfully in another container...
  const div = document.createElement("div");
  ReactDOM.render(<ProgressBar value="50" limit="150" />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders progressbar correctly", () => {
  // test that the component contains all correct elements required...
  const component = render(<ProgressBar value="50" limit="150" />);
  expect(component.getByTestId("mainDiv"));
  expect(component.getByTestId("progBarLabel"));
  expect(component.getByTestId("progressDiv"));
});

it("match snapshot", () => {
  // test that the component matches the snapshot,
  // creating a snpshot if it doesn't exist...
  const tree = renderer.create(<ProgressBar value="50" limit="150" />);
  expect(tree).toMatchSnapshot();
});
