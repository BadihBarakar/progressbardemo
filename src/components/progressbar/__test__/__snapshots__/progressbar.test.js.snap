// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="progress-bar"
  data-testid="mainDiv"
>
  <label
    data-testid="progBarLabel"
    style={
      Object {
        "left": "49%",
        "position": "absolute",
      }
    }
  >
    33.3%
  </label>
  <div
    className="progress-value"
    data-testid="progressDiv"
    style={
      Object {
        "backgroundColor": "red",
        "width": 500,
      }
    }
  />
</div>
`;
